Added Vagrantfile so that developers can spin up a VM to build geth

prasanna opened the pull request:
Added a Vagrantfile so that developers can spin up a VM to build geth.

Steps would be:
- Fork https://github.com/ethereum/go-ethereum
- ```cd <wherever-you-want-it>```
- ```git clone https://github.com/<user>/go-ethereum.git```
- ```cd go-ethereum```
- ```vagrant up```
- ```vagrant ssh```
- ```cd ~vagrant/go/src/github.com/ethereum/go-ethereum```
- ```make geth```

Runs ubuntu/trust64 and requires virtualbox and 2GB of RAM.

robotally commented:
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Sun Jan 24 02:42:14 UTC 2016_

obscuren commented on Vagrantfile:
```diff
@@ -0,0 +1,87 @@
+    sudo add-apt-repository -y ppa:ethereum/ethereum-dev
+    sudo apt-get update
+
+    sudo apt-get install -y build-essential libgmp3-dev golang git-all
```
You may remove the libgmp dep. It's no longer required by the `develop` branch :-)

karalabe commented:
@obscuren Just thinking here, should we maybe move all these container and VM images into a toplevel folder? We have one for docker, Vagrant could be arguable useful, and I guess there would be others coming eventually, so it would litter the codebase if each was dumped in the root.

Maybe something along the lines:

```
containers\docker\master
containers\docker\develop
containers\vagrant\...
```

The reason I'm more asking that stating is because our docker hub things are configured by caktux against the current layout, so chanign would mean updating them too. However it would be nice to do this in a future proof way. Ideas?

obscuren commented:
I'd be happy to see them moved @karalabe. @caktux care to weigh in?

karalabe commented:
I've already moved them when adding the Alpine images.

caktux commented:
Yep, the move was a good idea, so this would just need to go in `containers/vagrant/`

obscuren commented:
Awesome!

> On 03 Feb 2016, at 21:57, caktux <notifications@github.com> wrote:
> 
> Yep, the move was a good idea, so this would just need to go in containers/vagrant/
> 
> —
> Reply to this email directly or view it on GitHub.
>

karalabe commented:
@prasanna Could you please move the Vagrantfile into `containers/vagrant/`? Also the file contains quite a lot of "sample" and "explanation" comments. I don't think those are warranted to be checked into the repo. Just use a clean Vagrantfile with only the essentials, this isn't the place to teach people how to assemble a Vagrant image.

prasanna commented:
I did 3 things:
- Removed comments from Vagrantfile
- Removed libgmp dependency
- Moved Vagrantfile to containers/vagrant

You now have to cd into the containers/vagrant folder before running vagrant up. The setup steps now look like:

- Fork https://github.com/ethereum/go-ethereum
- cd <wherever-you-want-it>
- git clone https://github.com/<user>/go-ethereum.git
- cd go-ethereum/containers/vagrant
- vagrant up
- vagrant ssh
- cd ~vagrant/go/src/github.com/ethereum/go-ethereum
- make geth

I also merged the latest on the develop branch. Looks like the a couple of builds on it were broken, so they are still broken on this pull request.

Hope this works for everyone! Enjoy!

-Prasanna

obscuren commented:
Thanks, L]looks good to me now. Please squash your commit in to one single commit and please get rid of the merge commit. Instead you should rebase on develop :-)

prasanna commented:
Done!
